supported_os_controller.php

```diff
@@ -52,5 +56,217 @@ public function os()
+            return;
+        }
+
+        $queryobj = new Supported_os_model();
+
+        // Get YAML from supported_os GitHub
+        ini_set("allow_url_fopen", 1);
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, true);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_URL, 'https://raw.githubusercontent.com/munkireport/supported_os/master/supported_os_data.json');
+        $yaml_result = curl_exec($ch);
+
+        // Check if we got results
+        if (strpos($yaml_result, '"current_os":') === false ){
+            $yaml_result = file_get_contents(__DIR__ . '/supported_os_data.json');
+            $return_status = 2;
+            $cache_source = 2;
+        } else {
+            $return_status = 1;
+            $cache_source = 1;
+        }
+
+        $yaml_data = json_decode($yaml_result);
+        $current_os = $yaml_data->current_os;
+        $digits = explode('.', $current_os);
+        $mult = 10000;
+        $current_os = 0;
+        foreach ($digits as $digit) {
+            $current_os += $digit * $mult;
+            $mult = $mult / 100;
+        }
+
+        // Get the current time
+        $current_time = time();
+        
+        // Save new cache data to the cache table
+        munkireport\models\Cache::updateOrCreate(
+            [
+                'module' => 'supported_os', 
+                'property' => 'yaml',
+            ],[
+                'value' => $yaml_result,
+                'timestamp' => $current_time,
+            ]
+        );
+        munkireport\models\Cache::updateOrCreate(
+            [
+                'module' => 'supported_os', 
+                'property' => 'source',
+            ],[
+                'value' => $cache_source,
+                'timestamp' => $current_time,
+            ]
+        );
+        munkireport\models\Cache::updateOrCreate(
+            [
+                'module' => 'supported_os', 
+                'property' => 'current_os',
+            ],[
+                'value' => $current_os,
+                'timestamp' => $current_time,
+            ]
+        );
+        munkireport\models\Cache::updateOrCreate(
+            [
+                'module' => 'supported_os', 
+                'property' => 'last_update ',
+            ],[
+                'value' => $current_time,
+                'timestamp' => $current_time,
+            ]
+        );
+        
+        // Send result
+        $out = array("status"=>$return_status,"source"=>$cache_source,"timestamp"=>$current_time,"current_os"=>$current_os);
+        $obj->view('json', array('msg' => $out));
+    }
+
+     /**
+     * Pull in supported os data for all serial numbers :D
+     *
+     * @return void
+     * @author tuxudo
+     **/
+    public function pull_all_supported_os_data($incoming_serial = '')
+    {
+        $obj = new View();
+        // Authenticate
+        if (! $this->authorized()) {
```

Comment:
Please remove the authorised() block - not necessary anymore

Reply:
Removed

Reply:
Removed